uint32 explr_step
uint32 learning_ind
float32[] samples
int16[] samples_layout
float32[] distribution
